src/compile/data/dataflow.ts

```diff
@@ -116,11 +116,14 @@ export abstract class DataFlowNode {
     // remove old links
     this._children = []; // equivalent to removing every child link one by one
     parent.removeChild(this);
-    parent.parent.removeChild(parent);
+    const loc = parent.parent.removeChild(parent);
 
     // swap two nodes
-    this.parent = newParent;
-    parent.parent = this;
+    this._parent = newParent;
+    newParent.addChild(this, loc);
+
+    parent._parent = this;
+    this.addChild(parent, loc);
```

Comment:
I don't get this one. Isn't `loc` the location in `parent.parent`?

```suggestion
    parent.parent = this;
```

Reply:
I tested with this change and it seems like your examples are still correct.

Reply:
Thanks so much!  Yes, happy with your suggestion - that was just me not having a great grasp of some of the internals.